tests/integration/test_generic_upgrade.py

```diff
@@ -81,3 +79,15 @@ def verify_results(output_dir, num_jobs):
     for result in results:
         assert result.status == "finished"
         assert result.return_code == 0
+
+
+def verify_upgrade_results(output_dir):
```

Comment:
I added this function to check if all violations are resolved. Does this look okay?